pandas/io/formats/style_render.py

```diff
@@ -1391,13 +1406,23 @@ def _str_escape(x, escape):
     return x
 
 
+def _render_href(x):
+    """uses regex to detect a common URL pattern and converts to HTML <a> tag"""
+    if isinstance(x, str):
+        href = '<a href="{0}" target="_blank">{0}</a>'
+        pat = r"(https?:\/\/|ftp:\/\/|www.)[\w/\-?=%.]+\.[\w/\-&?=%.]+"
```

Comment:
Is there a way to somehow leverage `pandas.io.common.is_url`?

Reply:
I looked but I don't believe this finds URLs within a random string. Is why I used `re`. Maybe wrong.

Reply:
Hmm maybe you can leverage `is_url` to avoid false positives from the regex like

```
url = re.search(pat, x).group()
if pandas.io.common.is_url(url):
     ...
```

Reply:
Thats a nice idea, because the regex will permit some invalid urls, like `http://.1.2`, but the `is_url` function doesn't really help here since it detects that as a `True` url anyway, and it is more restrictive since it requires a `scheme` so `www.1.2` is apparently an invalid URL, according to `is_url`, since the scheme is not specified (a browser will default to using `https://` I believe). 

Happy if someone with more experience can improve this basic pattern search and url check, but I think as it stands is a good enough start

Reply:
Gotcha. Okay this regex approach is good for now.